snowflake/src/main/java/org/apache/iceberg/snowflake/entities/SnowflakeTableMetadata.java

```diff
@@ -0,0 +1,158 @@
+      String icebergMetadataLocation,
+      String status,
+      String rawJsonVal) {
+    this.snowflakeMetadataLocation = snowflakeMetadataLocation;
+    this.icebergMetadataLocation = icebergMetadataLocation;
+    this.status = status;
+    this.rawJsonVal = rawJsonVal;
+  }
+
+  /** Storage location of table metadata in Snowflake's path syntax. */
+  public String getSnowflakeMetadataLocation() {
+    return snowflakeMetadataLocation;
+  }
+
+  /** Storage location of table metadata in Iceberg's path syntax. */
+  public String getIcebergMetadataLocation() {
+    return icebergMetadataLocation;
+  }
+
+  public String getStatus() {
+    return status;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    } else if (!(o instanceof SnowflakeTableMetadata)) {
+      return false;
+    }
+
+    // Only consider parsed fields, not the raw JSON that may or may not be the original source of
+    // this instance.
+    SnowflakeTableMetadata that = (SnowflakeTableMetadata) o;
+    return Objects.equal(this.snowflakeMetadataLocation, that.snowflakeMetadataLocation)
+        && Objects.equal(this.icebergMetadataLocation, that.icebergMetadataLocation)
+        && Objects.equal(this.status, that.status);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(snowflakeMetadataLocation, icebergMetadataLocation, status);
+  }
+
+  @Override
+  public String toString() {
+    return String.format(
+        "snowflakeMetadataLocation: '%s', icebergMetadataLocation: '%s', status: '%s",
+        snowflakeMetadataLocation, icebergMetadataLocation, status);
+  }
+
+  /**
+   * Translates from Snowflake's path syntax to Iceberg's path syntax for paths matching known
+   * non-compatible Snowflake paths. Throws IllegalArgumentException if the prefix of the
+   * snowflakeLocation is a known non-compatible path syntax but fails to match the expected path
+   * components for a successful translation.
+   */
+  public static String getIcebergLocationFromSnowflakeLocation(String snowflakeLocation) {
+    if (snowflakeLocation.startsWith("azure://")) {
+      // Convert from expected path of the form:
+      // azure://account.blob.core.windows.net/container/volumepath
+      // to:
+      // wasbs://container@account.blob.core.windows.net/volumepath
+      Matcher matcher = SNOWFLAKE_AZURE_PATTERN.matcher(snowflakeLocation);
+      Preconditions.checkArgument(
+          matcher.matches(),
+          "Location '%s' failed to match pattern '%s'",
+          snowflakeLocation,
+          SNOWFLAKE_AZURE_PATTERN);
+      return String.format(
+          "wasbs://%s@%s/%s", matcher.group(2), matcher.group(1), matcher.group(3));
+    } else if (snowflakeLocation.startsWith("gcs://")) {
+      // Convert from expected path of the form:
+      // gcs://bucket/path
+      // to:
+      // gs://bucket/path
+      return "gs" + snowflakeLocation.substring(3);
+    }
+    return snowflakeLocation;
```

Comment:
Style: there should be an empty line between control flow blocks and the following statement.

Reply:
Done